### PR TITLE
Get jobs enhancements

### PR DESCRIPTION
**Pull request by Toilal**
See https://github.com/taskforcesh/bullmq/issues/1112

The PR is splitted in 3 incremental commits. Each commit brings more changes, so you can choose what to merge or not.

First two commits should be safe to merge. 3rd commit improves consistency and should ensure that `getJobs` always returns the same count as `getJobsCount`, but it's more intrusive.

### CODE REVIEW

**Comment by ATheCoder** [comment]
@manast

**Review by manast — APPROVED** [review]

File: src/classes/queue-getters.ts [hunk 1]
```diff
@@ -45,16 +45,40 @@ export class QueueGetters<
...
+    if (typeof types === 'string') {
+      types = [types];
+    }
+
+    if (Array.isArray(types) && types.length > 0) {
+      const sanitizedTypes = [...types];
+
+      if (
+        sanitizedTypes.indexOf('waiting') !== -1 ||
+        sanitizedTypes.indexOf('waiting-children')
```
Comment (roggervalf) [inline src/classes/queue-getters.ts:58]: we are not considering paused when waiting-children is provided in the current master branch, you could update your branch with master

File: src/classes/queue-getters.ts [hunk 2]
```diff
@@ -45,16 +45,40 @@ export class QueueGetters<
     });
   }
 
+  private sanitizeJobTypes(types: JobType[] | JobType | undefined): JobType[] {
+    if (typeof types === 'string') {
```
Comment (roggervalf) [inline src/classes/queue-getters.ts:49]: what about:
`const currentTypes = typeof types === 'string' ? [types] : types;`

**Review by roggervalf — CHANGES_REQUESTED** [review]

File: src/classes/queue-getters.ts (hunk 2 above)
Comment (Toilal) [reply src/classes/queue-getters.ts:49]: Il would create `[undefined]` array if parameter is undefined

File: src/classes/queue-getters.ts (hunk 1 above)
Comment (Toilal) [reply src/classes/queue-getters.ts]: I have removed `sanitizedTypes.indexOf('waiting-children')` from this condition

File: src/classes/queue-getters.ts [hunk 3]
```diff
@@ -69,12 +75,7 @@ export class QueueGetters<
     Returns the number of jobs waiting to be processed.
   */
   count(): Promise<number> {
-    return this.getJobCountByTypes(
-      'waiting',
-      'paused',
-      'delayed',
-      'waiting-children',
-    );
+    return this.getJobCountByTypes('waiting', 'delayed', 'waiting-children');
```
Comment (Toilal) [inline src/classes/queue-getters.ts:78]: I'm not sure if 'paused' should be removed here

Comment (manast) [reply src/classes/queue-getters.ts:78]: It should not be removed. When the queue is paused what happens is that the wait key gets renamed to "paused", but other than that the status of a job is still waiting.

Comment (Toilal) [reply src/classes/queue-getters.ts]: Thanks, I have rollbacked this method.

File: src/classes/queue-getters.ts (hunk 2 above)
Comment (roggervalf) [reply src/classes/queue-getters.ts:49]: hey @Toilal, but if types is `undefined`, `typeof types === 'string'` would be `false`, in that case currentTypes will be `undefined`, not `[undefined]` :thinking:

Comment (Toilal) [reply src/classes/queue-getters.ts:49]: Ho yes my mistake, I have updated the PR :)

**Review by roggervalf — APPROVED** [review]
LGTM

**Pull request merged**